flowplayer.hlsjs.js

```diff
@@ -602,7 +602,20 @@
                                             maxLevel = hls.loadLevel;
                                         }
                                         break;
-
+                                    case "FRAG_PARSING_METADATA":
+                                        data.samples.forEach(function(sample) {
+                                          var metadataHandler = function() {
+                                            if (videoTag.currentTime < sample.dts) return;
+                                            videoTag.removeEventListener('timeupdate', metadataHandler);
+                                            var raw = new TextDecoder('utf-8').decode(sample.data);
+                                            player.trigger('metadata', [player, {
+                                              key: raw.substr(10, 4),
+                                              data: raw.substr(21)
+                                            }]);
+                                          };
+                                          videoTag.addEventListener('timeupdate', metadataHandler);
```

Comment:
What is out of scope? Re-reading  your comment this can be made with bean too.  I guess you are removing all events on unload and want to have a shorthand for that? I'm creating own handler for all ID3 events and after it is fired it removes itself. So the handler must be declared within the loop.